### PR TITLE
Mrshu/sticky notes

### PR DESCRIPTION
**Pull request by mrshu**
This PR implements the first version of Sticky Notes as described in #42.

### CODE REVIEW

**Comment by mrshu** [comment]
to @pepol 

cc @JakubNvk @Adman

File: sticky_notes.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        return "Sticky note for {0} prepared.".format(user)
+
+    @event
+    def send_notes(event):
+        format_note = lambda x, y: "{0}: {1}".format(x, y)
+
+        if event.event_type == 'join':
+            nick = event.meta['nick']
+            if nick in self.notes:
+                last_note = self.notes[nick].pop()
```
Comment (pepol) [inline sticky_notes.py]: This is kind of duplicate code. While I see the point, wouldn't it be better to write all the notes in the loop and then return something like: "End of log for {username}." ?

**Comment by Adman** [comment]
@mrshu shouldn't be the notes sent privately to user?

**Comment by pepol** [comment]
@adman what's the point? They will be written in channel history anyway (when issuing a command).

**Comment by JakubNvk** [comment]
@pepol :+1:

**Comment by Adman** [comment]
@pepol yep, they will. I was just wondering whether it won't be spamming others.

**Comment by mrshu** [comment]
@Adman the point of this is to notify the user of something when they join a channel. For people who have bouncers and stuff this doesn't mean much anyway so while I get your point with spamming the channel I do not think it will be a bit issue.

File: sticky_notes.py (hunk 1 above)
Comment (mrshu) [inline sticky_notes.py]: While I get what you are saying, the way in which brutal is structured would be much more OK with just returning some messages as a reply than forcing another virtual reply event for via `self.msg`.

**Comment by pepol** [comment]
@mrshu LGTM. Merging this.

**Pull request merged**